File: sudoku-auth/auth-boot/src/main/java/com/zumin/sudoku/auth/AuthBootApplication.java
package com.zumin.sudoku.auth;

import com.zumin.sudoku.admin.feign.UserFeign;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
import org.springframework.cloud.openfeign.EnableFeignClients;

@EnableDiscoveryClient
@EnableFeignClients(basePackageClasses = {UserFeign.class})
@SpringBootApplication(scanBasePackages = {"com.zumin.sudoku.auth", "com.zumin.sudoku.common"})
public class AuthBootApplication {

  public static void main(String[] args) {
    SpringApplication.run(AuthBootApplication.class, args);
  }

}

File: sudoku-auth/auth-boot/src/main/java/com/zumin/sudoku/auth/AuthBootApplication.kt
package com.zumin.sudoku.auth

import com.zumin.sudoku.ums.UserFeign
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.cloud.client.discovery.EnableDiscoveryClient
import org.springframework.cloud.openfeign.EnableFeignClients

@SpringBootApplication
@EnableDiscoveryClient
@EnableFeignClients(basePackageClasses = [UserFeign::class])
class AuthBootApplication

fun main(args: Array<String>) {
  runApplication<AuthBootApplication>(*args)
}
